src/agent/scm/git.ts

```diff
@@ -103,19 +113,36 @@ export class GitScmProvider extends scmm.ScmProvider {
         return Q(0)
         .then((code: number) => {
 	        if (!this.enlistmentExists()) {
-	        	return this.gitw.clone(giturl, true, folder, gopt);
+	        	return this.gitw.clone(giturl, true, folder, gopt).then((result: number) => {
+                    if (isPullRequest) {
+                        // clone doesn't pull the refs/pull namespace, so fetch it
+                        return this.gitw.fetch(['origin', srcBranch], gopt);
```

Comment:
Does this work if the enlistment doesn't exist?  I thought if there's no folder, you would have to do mkdir, git init, and then fetch ...  At one point, we considered doing that instead of clone.